learning/tour-of-beam/frontend/lib/components/profile/user_menu.dart

```diff
@@ -125,9 +125,29 @@ class _Buttons extends StatelessWidget {
         ),
         const BeamDivider(),
         _IconLabel(
-          onTap: () {},
+          onTap: () {
+            closeOverlayCallback();
+            showDialog(
+              context: context,
+              builder: (context) => Dialog(
+                backgroundColor: Colors.transparent,
+                child: BeamAlertDialog(
+                  body: 'dialogs.deleteAccountWarning'.tr(),
+                  continueLabel: 'ui.deleteMyAccount'.tr(),
+                  title: 'ui.deleteTobAccount'.tr(),
+                  onContinue: () {
+                    authNotifier.deleteAccount().then(
+                      (_) {
+                        Navigator.pop(context);
+                      },
+                    );
+                  },
```

Comment:
1. async / await.
2. Show progress?

Reply:
1. Oh, by the way this is just the same problem, but the linter cannot spot it. One solution is to let the dialog close itself by passing a callback to `onContinue`, but this is ugly. Try to think of better ideas. Also this is exactly why in app_state we have states pushing themselves out without a context.
2. Yes.

Reply:
The simplest solution is to first pop the dialog and then to delete the account.